src/lib/input/input-container.scss

```diff
@@ -237,8 +237,16 @@ $mat-input-underline-disabled-background-image:
   }
 }
 
-// Prevents the prefix and suffix from stretching together with the container.
 .mat-input-prefix, .mat-input-suffix {
+  // Prevents the prefix and suffix from stretching together with the container.
   width: 0.1px;
   white-space: nowrap;
+
+  // Allow icons in a prefix/suffix to adapt to the correct size.
+  & .mat-icon {
+    width: auto;
+    height: auto;
+    font-size: 100%;
+    vertical-align: top;
```

Comment:
It was part of the styles that were already in the demo app css that i was migrating into the library css. I'm not sure on the original rationale for `top`, but leaving it as `baseline` causes the prefix/suffix element to be too tall, `bottom` and `middle` also fix the issue but I just left `top` since that was in the demo css already.